src/nn/mod.rs

```diff
@@ -104,6 +104,10 @@
 //! mlp.load_state_dict(state_dict)
 //! ```
 
+mod num_params;
+mod reset_params;
+pub mod visitors;
```

Comment:
Come up with a better name than "visitors". The real purpose for users will be getting access to TensorCollection to impl for modules. Perhaps "collections"?